src/coord/src/catalog/migrate.rs

```diff
@@ -94,6 +96,106 @@ pub(crate) fn migrate(catalog: &mut Catalog) -> Result<(), anyhow::Error> {
 // AST migrations -- Basic AST -> AST transformations
 // ****************************************************************************
 
+/// Rewrites all references of `pg_catalog.char` to `pg_catalog.text`, which
+/// matches the previous char implementation's semantics.
+///
+/// Note that the previous `char` "implementation" was simply an alias to
+/// `text`. However, the new `char` semantics mirrors Postgres' `bpchar` type,
+/// which `char` is now essentially an alias of.
+///
+/// This approach is safe because all previous references to `char` were
+/// actually `text` references. All `char` references going forward will
+/// properly behave as `bpchar` references.
+fn ast_rewrite_pg_catalog_char_to_text_0_9_1(
+    stmt: &mut sql::ast::Statement<Raw>,
+) -> Result<(), anyhow::Error> {
+    struct TypeNormalizer;
+
+    lazy_static! {
+        static ref CHAR_REFERENCE: UnresolvedObjectName =
+            UnresolvedObjectName(vec![Ident::new(PG_CATALOG_SCHEMA), Ident::new("char")]);
+        static ref TEXT_REFERENCE: UnresolvedObjectName =
+            UnresolvedObjectName(vec![Ident::new(PG_CATALOG_SCHEMA), Ident::new("text")]);
+    }
+
+    impl<'ast> VisitMut<'ast, Raw> for TypeNormalizer {
+        fn visit_data_type_mut(&mut self, data_type: &'ast mut DataType<Raw>) {
+            if let DataType::Other { name, .. } = data_type {
+                if name.name() == &*CHAR_REFERENCE {
+                    let t = TEXT_REFERENCE.clone();
+                    *name = match name {
+                        RawName::Name(_) => RawName::Name(t),
```

Comment:
I am afraid this rewrites ```CHAR(N)``` into ```TEXT(N)``` , which is not  valid and the server refuses to start up:
```
materialized: cannot migrate from catalog version pre-v0.8.4 to version 0.9.1-dev (earlier versions might still work): corrupt catalog: failed to deserialize item u74 (materialize.public.char_view): pg_catalog.text does not support type modifiers
```